sdks/python/src/opik/integrations/sagemaker/auth.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import os
+import httpx
+
+from typing import Any
+
+import opik.hooks
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _in_aws_sagemaker() -> bool:
+    return os.getenv("AWS_PARTNER_APP_AUTH") is not None
+
+
+class SagemakerAuth(httpx.Auth):
+    def __init__(self, auth_provider: Any) -> None:
+        self.auth_provider = auth_provider
+
+    def auth_flow(self, request):  # type: ignore
+        if not _in_aws_sagemaker():
+            yield request
+
+        url, signed_headers = self.auth_provider.get_signed_request(
+            request.url, request.method, request.headers, request.content
+        )
+
+        request.url = url
+        request.headers.update(signed_headers)
+
+        yield request
+
+
+def _setup_aws_sagemaker_session_hook() -> None:
+    import sagemaker
```

Comment:
@alexkuzmik since we dont depend on this SDK directly, lets to try catch and promt the user to install this is they get here without pip install

Reply:
@Nimrod007 I assumed it's installed in the Sagemaker environment.
I'll add the check.